bentoml/_internal/frameworks/transformers.py

```diff
@@ -106,427 +65,209 @@ def check_flax_supported() -> None:  # pragma: no cover
-        pipeline or a tuple containing :obj:`PretrainedConfig`, :obj:`Model` class object defined by :obj:`transformers`, with an optional :obj:`Tokenizer` class, or :obj:`FeatureExtractor` class for the given model saved in BentoML modelstore.
-
-    Examples:
+try:
+    import transformers
 
-    .. code-block:: python
+    _check_flax_supported()
+except ImportError:  # pragma: no cover
+    raise MissingDependencyException(
+        """\
+        transformers is required in order to use module `bentoml.transformers`.
+        Instruction: Install transformers with `pip install transformers`.
+        """
+    )
 
-        import bentoml
-        model, tokenizer = bentoml.transformers.load('custom_gpt2')
 
-    If you want to returns an config object:
+class TransformersOptions(ModelOptions):
+    """Options for the Transformers model."""
 
-    .. code-block:: python
+    def __init__(self, **kwargs):
+        class _Schema(pydantic.BaseModel):
+            task: str
+            pipeline: bool = True
 
-        import bentoml
-        config, model, tokenizer = bentoml.transformers.load('custom_gpt2', return_config=True, tokenizer_kwargs={"use_fast":True})
+        try:
+            _Schema(**kwargs)
+        except pydantic.ValidationError:
+            raise BentoMLException(f"Model options {kwargs} is not valid.")
 
-    If the pipeline is saved with :code:`bentoml.transformers.save()`, then :code:`load()` will return pipeline objects:
+        super().__init__(**kwargs)
 
-    .. code-block:: python
 
-        import bentoml
-        pipeline = bentoml.transformers.load("roberta_text_classification", return_all_scores=True)
-    """  # noqa
-    check_flax_supported()  # pragma: no cover
-    model = model_store.get(tag)
+def get(tag_like: str | Tag) -> Model:
+    model = bentoml.models.get(tag_like)
     if model.info.module not in (MODULE_NAME, __name__):
-        raise BentoMLException(
-            f"Model {tag} was saved with module {model.info.module}, failed loading with {MODULE_NAME}."
+        raise NotFound(
+            f"Model {model.tag} was saved with module {model.info.module}, failed loading with {MODULE_NAME}."
         )
+    model.info.parse_options(TransformersOptions)
+    return model
 
-    if model.info.context["pipeline"]:
-        _tasks = model.info.context["task"]
-        return transformers.pipeline(_tasks, model.path, **kwargs)
-    else:
-        config_kwargs = kwargs.pop("config_kwargs", {})
-        config: "ext.PretrainedConfig" = AutoConfig.from_pretrained(
-            model.path, **config_kwargs
-        )
 
-        model_kwargs = kwargs.pop("model_kwargs", {})
+def load_model(
+    bento_model: str | Tag | Model,
+    **kwargs: t.Any,
+) -> "ext.TransformersPipeline":
```

Comment:
actually, since you import `from __future__ import annotations`, no need to quote this `ext.TransformersPipeline` anymore